extensions/amp-video/1.0/video-wrapper.js

```diff
@@ -106,14 +109,25 @@ export function VideoWrapper({
 
   useLayoutEffect(() => {
     if (mediasession && playing && metadata) {
-      setMediaSession(window, metadata, play, pause);
+      clearMediaSession.current = setMediaSession(
+        window,
+        metadata,
+        play,
+        pause
+      );
     }
-    return () => {
-      // TODO(alanorozco): Clear media session.
-      // (Tricky because we don't want to clear a different active session.)
-    };
   }, [mediasession, playing, metadata, play, pause]);
 
+  useEffect(
+    // Clear only on unmount.
+    () => () => {
```

Comment:
Nit: I find arrows returning arrows to be really difficult to parse mentally. Can we just break separate this into an explicit return statement?